.github/workflows/tests.yml

```diff
@@ -11,48 +11,35 @@ defaults:
     shell: bash
 
 jobs:
-  test-check:
-    name: trivy (github-check)
+  setup:
     runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.output-matrix.outputs.matrix }}
 
     steps:
       - uses: actions/checkout@v4
-
-      - uses: ./
-        continue-on-error: true
-        id: test
-        with:
-          github_token: ${{ secrets.github_token }}
-          trivy_command: config
-          trivy_target: .
-          reporter: github-check
-          level: info
-          working_directory: testdata/with_detections
-
-      # The check is expected to fail on the test data
-      - name: Check return codes
-        if: success() || failure ()
+      - name: Set up matrix
+        id: output-matrix
         run: |
-          trivy_return="${{ steps.test.outputs.trivy-return-code }}"
-          reviewdog_return="${{ steps.test.outputs.reviewdog-return-code }}"
-
-          if [[ "$trivy_return" -eq 1 ]]; then
-            echo "trivy correctly returned failure ${trivy_return}"
-          else
-            echo "trivy returned ${trivy_return}, expected '1'. Failing..."
-            exit 1
-          fi
+          # Find all the directories in the testdata directory
+          DIRS="$(find ./testdata -type d)"
+          for dir in ${DIRS}; do
+            if [ "$(echo "${DIRS}" | grep -c "${dir}" || true)" -le 1 ]; then
+              DIRS_RESULT="${DIRS_RESULT}${dir},"
+            fi
+          done
+          MATRIX_JSON=$(echo "${DIRS_RESULT}" | sed 's/,$//' | sed 's/\\n//g' | jq -R -s -c 'split(",") | [.[] | {dir: ., command: split("/")[2], type: split("/")[4], label: [split("/")[2],split("/")[3]] | join("/")}]')
```

Comment:
Can you add some comments? 
I don't quite understand what this script is doing.